scipy/optimize/_constraints.py

```diff
@@ -250,6 +266,15 @@ def _input_validation(self):
         except ValueError:
             message = "`lb`, `ub`, and `keep_feasible` must be broadcastable."
             raise ValueError(message)
+        try:
+            lb = self.lb.astype(np.double)
+            ub = self.ub.astype(np.double)
+        except (ValueError, TypeError) as exc:
+            message = ("Lower and upper bounds must be reals")
+            raise ValueError(message) from exc
+
+        if (self.lb > self.ub).any():
```

Comment:
I personally would like different behaviour here. Something along the lines of

```
_lb = np.minimum(self.lb, self.ub)
_ub = np.maximum(self.lb, self.ub)
self.lb = _lb
self.ub = _ub
```

It's more robust as less input validation is required for a downstream application. However, it may not be to everyone's taste.

Reply:
Can you bring this point up on the scipy-dev mailing list, I think it'll need a discussion.

Reply:
This is code moved from `_validate_bounds` in [scipy/optimize/_minimize.py](https://github.com/scipy/scipy/pull/18483/files#diff-61ffae76b40123d1acc8d871de22c1fbebdd49a8c1d90a7933e985a66a9f19a8), without change. 

For discussing a new, different behavior, such as the automagical conversion you describe, maybe it would make sense to open a new issue or a thread on the mailing list.